cpp/src/gandiva/precompiled/extended_math_ops_test.cc

```diff
@@ -87,6 +87,20 @@ TEST(TestExtendedMathOps, TestLogWithBase) {
   EXPECT_EQ(context1.has_error(), false);
 }
 
+TEST(TestExtendedMathOps, TestRoundDecimal) {
+  EXPECT_FLOAT_EQ(round_float32_int32(1234.789, 2), 1234.79);
+  EXPECT_FLOAT_EQ(round_float32_int32(1234.12345, -3), 1000);
+  EXPECT_FLOAT_EQ(round_float32_int32(-1234.4567, 3), -1234.457);
+  EXPECT_FLOAT_EQ(round_float32_int32(-1234.4567, -3), -1000);
+  EXPECT_FLOAT_EQ(round_float32_int32(1234.4567, 0), 1234);
+
+  VerifyFuzzyEquals(round_float64_int32(1234.789, 2), 1234.79);
```

Comment:
What happens when we convert (double)INT_MAX+1 or (double)INT_MIN-1 int32?

Reply:
I am sorry, I did not fully understand your concern. Are you suggesting that some unit tests be added for `round_float64_int32()` where the first argument is INT_MAX+1 or INT_MIN-1?

Reply:
Sure, there are examples. I do not know the results in `int32`.

```
VerifyFuzzyEquals(round_float64_int32((double)INT_MAX+1, 0), ???);
VerifyFuzzyEquals(round_float64_int32((double)INT_MIN-1, 0), ???);

```

Reply:
Added the unit tests.